From df5a548fcd64c2b98ed79adbbf17d876548f15ee Mon Sep 17 00:00:00 2001
From: Hien Huynh <hien.huynh.px@renesas.com>
Date: Tue, 30 Jun 2020 14:39:35 +0700
Subject: [PATCH 189/458] arm64: dts: renesas: hihope-common: Enable Audio
 support in Low speed connector

This patch enables sound support in Low speed connector
for the common HiHope G2M and G2N boards by adding
device node of the PCM5102 codec (External I2S Board
in Low speed Connector).

Therefore, we support 2 audio codecs in common plarform:
1. HDMI audio from built-in HDMI transmitter. (default)
2. External I2S Audio Codec PCM5102A in Low Speed Connector.

Signed-off-by: Hien Huynh <hien.huynh.px@renesas.com>
---
 arch/arm64/boot/dts/renesas/hihope-common.dtsi | 54 +++++++++++++++++++++-----
 1 file changed, 45 insertions(+), 9 deletions(-)

diff --git a/arch/arm64/boot/dts/renesas/hihope-common.dtsi b/arch/arm64/boot/dts/renesas/hihope-common.dtsi
index 04823d1..04f9578 100644
--- a/arch/arm64/boot/dts/renesas/hihope-common.dtsi
+++ b/arch/arm64/boot/dts/renesas/hihope-common.dtsi
@@ -89,7 +89,8 @@
 
 		label = "rcar-sound";
 
-		dais = <&rsnd_port>;
+		dais = <&rsnd_port0	/* HDMI Audio */
+			&rsnd_port1>;	/* External PCM5102A Codec */
 	};
 
 	vbus0_usb2: regulator-vbus0-usb2 {
@@ -142,6 +143,17 @@
 	vspm_if {
 		compatible = "renesas,vspm_if";
 	};
+
+	pcm5102a: pcm5102a {
+		compatible = "ti,pcm5102a";
+		#sound-dai-cells = <0>;
+
+		port {
+			pcm5102a_endpoint: endpoint {
+				remote-endpoint = <&rsnd_endpoint1>;
+			};
+		};
+	};
 };
 
 &audio_clk_a {
@@ -319,6 +331,11 @@
 		function = "audio_clk";
 	};
 
+	sound_pins: sound {
+		groups = "ssi01239_ctrl", "ssi0_data", "ssi1_data_a";
+		function = "ssi";
+	};
+
 	usb0_pins: usb0 {
 		groups = "usb0";
 		function = "usb0";
@@ -353,7 +370,7 @@
 };
 
 &rcar_sound {
-	pinctrl-0 = <&sound_clk_pins>;
+	pinctrl-0 = <&sound_clk_pins &sound_pins>;
 	pinctrl-names = "default";
 
 	status = "okay";
@@ -361,19 +378,38 @@
 	/* Single DAI */
 	#sound-dai-cells = <0>;
 
-	rsnd_port: port {
-		rsnd_endpoint: endpoint {
-			remote-endpoint = <&dw_hdmi0_snd_in>;
+	ports {
+		rsnd_port0: port@0 {
+			rsnd_endpoint: endpoint {
+				remote-endpoint = <&dw_hdmi0_snd_in>;
+
 
-			dai-format = "i2s";
-			bitclock-master = <&rsnd_endpoint>;
-			frame-master = <&rsnd_endpoint>;
+				dai-format = "i2s";
+				bitclock-master = <&rsnd_endpoint>;
+				frame-master = <&rsnd_endpoint>;
 
-			playback = <&ssi2>;
+				playback = <&ssi2>;
+			};
+		};
+
+		rsnd_port1: port@1 {
+			rsnd_endpoint1: endpoint {
+				remote-endpoint = <&pcm5102a_endpoint>;
+
+				dai-format = "i2s";
+				bitclock-master = <&rsnd_endpoint1>;
+				frame-master = <&rsnd_endpoint1>;
+
+				playback = <&ssi1 &src1 &dvc0>;
+			};
 		};
 	};
 };
 
+&ssi1 {
+	shared-pin;
+};
+
 &rwdt {
 	timeout-sec = <60>;
 	status = "okay";
-- 
2.7.4

